core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -5919,6 +5921,30 @@ private Object coerce(Type sourceType, Object value, Type targetType)
         }
     }
 
+    private static RefreshType getRefreshType(Query query)
+    {
+        if (query.getQueryBody() instanceof QuerySpecification querySpecification) {
+            if (querySpecification.getGroupBy().isEmpty()
+                    && querySpecification.getWindows().isEmpty()
+                    && querySpecification.getFrom().stream().allMatch(from -> from instanceof Table)
```

Comment:
Table can also be a nested view. TBH I think we should decide on refresh type during logical planning because:
1. we know how to process `PlanNodes` better and we have all functions, tables and types resolved.
2. In case of recursive incremental query, we would need to compile such query (if possible) during planning anyway and it's hard to decouple compiling of incremental query from deciding if it's even possible to compile such query.